bool request

-----
float32[] vb
float32[] phi

